crates/astria-sequencer/src/mempool.rs

```diff
@@ -72,6 +84,7 @@ impl EnqueuedTransaction {
             tx_hash: signed_tx.sha256_of_proto_encoding(),
             signed_tx: Arc::new(signed_tx),
             address,
+            time_first_seen: Instant::now(),
```

Comment:
This is probably better suited to live in the `TransactionPriority` struct since it might be considered in the future as part of the prioritisation algo, and also that would keep `EnqueuedTransaction`'s hash a genuine way to determine equality/ordering.